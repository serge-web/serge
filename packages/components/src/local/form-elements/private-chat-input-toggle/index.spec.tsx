/* global it expect */

import React from 'react'
import renderer from 'react-test-renderer'

import PrivateChatInputToggle from './index'

const createNodeMock = (el: any): HTMLTextAreaElement | null => {
  if (el.type === 'textarea') {
    return document.createElement('textarea')
  }
  return null
}

const privatValue = (value: string): string => value

describe('PrivateChatInputToggle component:', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <PrivateChatInputToggle privatValue={privatValue} placeholder={'Enter your private message'} />,
        { createNodeMock }
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
