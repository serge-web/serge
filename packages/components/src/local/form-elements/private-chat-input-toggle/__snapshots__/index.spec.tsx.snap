// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateChatInputToggle component: renders correctly 1`] = `
Array [
  <div
    className="MuiBox-root MuiBox-root-1"
    onClick={[Function]}
  >
    <span
      className="private-badge-wrapper"
    >
      <span>
        <svg
          className="svg-inline--fa fa-paperclip"
        />
      </span>
      <span
        className="icon-private"
      >
        <svg
          className="svg-inline--fa fa-user-secret"
        />
      </span>
      <span
        className="label-private"
      >
        Private message
      </span>
    </span>
  </div>,
  <div
    className="MuiBox-root MuiBox-root-2 wrap-detail wrap-detail-opened"
  >
    <section
      className="input-container disable-offset fullwidth"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        onKeyDown={[Function]}
      >
        <div
          className="MuiInputBase-root MuiFilledInput-root makeStyles-root-3 MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiFilledInput-multiline makeStyles-multiline-5"
          onClick={[Function]}
        >
          <textarea
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiFilledInput-input makeStyles-input-4 MuiInputBase-inputMultiline MuiFilledInput-inputMultiline"
            disabled={false}
            name=""
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Enter your private message"
            required={false}
            rows={2}
            style={
              Object {
                "height": 0,
                "overflow": "hidden",
              }
            }
            value="private"
          />
          <textarea
            aria-hidden={true}
            className="MuiInputBase-input MuiFilledInput-input makeStyles-input-4 MuiInputBase-inputMultiline MuiFilledInput-inputMultiline"
            readOnly={true}
            style={
              Object {
                "height": 0,
                "left": 0,
                "overflow": "hidden",
                "position": "absolute",
                "top": 0,
                "transform": "translateZ(0)",
                "visibility": "hidden",
              }
            }
            tabIndex={-1}
          />
        </div>
      </div>
    </section>
  </div>,
]
`;
